### PR TITLE
Adding access patterns doc

### PR DESCRIPTION
**Pull request by codyseibert**
adding an access pattern document and .csv file so that people can get a better high level overview of the PK/SK/GS1PKs we use for our various data models.

### CODE REVIEW

File: docs/ACCESS_PATTERNS.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+| associate user with pending case                           | user\|${USER_ID}                                | pending-case\|${CASE_ID}                 |                                             |
+| a case                                                     | case\|${CASE_ID}                                | case\|${CASE_ID}                         |                                             |
+| how we store the list of trial sessions eligble for a case | eligible-for-trial-case-catalog                 | LasVegasNevada-H-B-20190816132910-107-19 | eligible-for-trial-case-catalog\|${CASE_ID} |
+| docket number generator counter                            | docketNumberCounter-2021                        | docketNumberCounter-${YEAR}              |                                             |
+| associate a irs practitioner onto a case                   | case\|${CASE_ID}                                | irsPractitioner\|${USER_ID}              |                                             |
```
Comment (matthopson) [inline docs/ACCESS_PATTERNS.md]: "an" irs practitioner

File: docs/ACCESS_PATTERNS.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+| save web socket connection associated with the user        | user\|${USER_ID}                                | connection\|${SOCKET_ID}                 | connection\|${SOCKET_ID}                    |
+| add a hearing to a case                                    | case\|${CASE_ID}                                | hearing\|${TRIAL_SESSION_ID}             |                                             |
+| a trial session record                                     | trial-session\|${TRIAL_SESSION_ID}              | trial-session\|${TRIAL_SESSION_ID}       | trial-session-catalog                       |
+| copy the trial session and attach it to the user           | trial-session-working-copy\|${TRIAL_SESSION_ID} | user\|${USER_ID}                         |                                             |
+| add a case not to the user                                 | user-case-note\|${USER}                         | user\|${USER_ID}                         |                                             |
```
Comment (matthopson) [inline docs/ACCESS_PATTERNS.md]: "note"

**Review by matthopson — APPROVED** [review]
Couple small typos (I didn't note them in the csv, but they're there too)

File: docs/ACCESS_PATTERNS.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+| associate user with a case                                 | user\|${USER_ID}                                | case\|${CASE_ID}                         | user-case\|${CASE_ID}                       |
+| associate user with pending case                           | user\|${USER_ID}                                | pending-case\|${CASE_ID}                 |                                             |
+| a case                                                     | case\|${CASE_ID}                                | case\|${CASE_ID}                         |                                             |
+| how we store the list of trial sessions eligble for a case | eligible-for-trial-case-catalog                 | LasVegasNevada-H-B-20190816132910-107-19 | eligible-for-trial-case-catalog\|${CASE_ID} |
+| docket number generator counter                            | docketNumberCounter-2021                        | docketNumberCounter-${YEAR}              |                                             |
```
Comment (mmarcotte) [inline docs/ACCESS_PATTERNS.md]: I guess the PK would also be:
```
docketNumberCounter-${YEAR}   
```

File: docs/ACCESS_PATTERNS.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+| user outbox workitems sorted by date                       | user-outbox\|${USER_ID}                         | $datetime                                | work-item\|${WORK_ITEM_ID}                  |
+| a work item entry                                          | work-item\|${WORK_ITEM_ID}                      | work-item\|${WORK_ITEM_ID}               | work-item\|${WORK_ITEM_ID}                  |
+| a work item on a case                                      | case\|${CASE_ID}                                | work-item\|${WORK_ITEM_ID}               |                                             |
+| associate practitioner by name for lookup                  | privatePractitioner\|${NAME}                    | user\|${userId}                          |                                             |
+| associate practitioner by bar number for lookup            | privatePractitioner\|${BAR_NUMBER}              | user\|${userId}                          |                                             |
```
Comment (mmarcotte) [inline docs/ACCESS_PATTERNS.md]: Love the list. I just wonder if you could sort by PK. I think everything falls into order better for my 🧠

**Review by mmarcotte — CHANGES_REQUESTED** [review]
just wondering if you could address the above, thanks Cody!

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**